luarules/gadgets/unit_zombies.lua

```diff
@@ -399,5 +391,27 @@ end
 function gadget:Initialize()
 	mapWidth = Game.mapSizeX
 	mapHeight = Game.mapSizeZ
-	initiationFrame = spGetGameFrame() + 1 --to avoid race conditions
+	gameFrame = spGetGameFrame()
+	local units = spGetAllUnits()
+	for _, unitID in ipairs(units) do
+		local identifiedZombie = spGetUnitRulesParam(unitID, "zombie")
+		if identifiedZombie and identifiedZombie == IS_ZOMBIE then
+			local unitDefID = spGetUnitDefID(unitID)
+			zombieWatch[unitID] = unitDefID
+			spGiveOrderToUnit(unitID, CMD_REPEAT, 1, 0)
+			spGiveOrderToUnit(unitID, CMD_MOVE_STATE, 2, 0)
+			spGiveOrderToUnit(unitID, CMD_FIRE_STATE, 2, 0)
+			local energyToSet = ENERGY_INCOME_MULTIPLIER * UnitDefs[unitDefID].energyCost
+			local metalToSet = METAL_INCOME_MULTIPLIER * UnitDefs[unitDefID].metalCost
+			spSetUnitResourcing(unitID, {["umm"] = metalToSet, ["ume"] = energyToSet})
```

Comment:
This duplicates a lot of the functionality in `spawnZombie`, it should be extracted to a common function.